src/util/Undefined.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Narrows `T` to a value which is not `null` or `undefined`,
+ * or throws an error if it is `null` or `undefined`.
+ */
+export function throwIfNullish<T>(t: T | null | undefined): T {
+  if (t === null || t === undefined) {
+    throw new Error('Expected non-nullish value!');
+  }
+  return t;
+}
```

Comment:
lol you should just pester the typescript maintainers to add this to the language.
maybe something like this would work:
```
const myVariable: string! = 'some string'
```
Where the `!` means it will be of type `string | never` because it throws for nullish values

Reply:
I need a runtime assertion, though -- your language suggestion looks like it'd be type-level only. It _could_ compile to something else, but TypeScript generally tries to be a superset of JavaScript.

Honestly we should just maintain our own TypeScript standard library